lib/iiif/image_info_extractor.rb

```diff
@@ -2,16 +2,36 @@ module IIIF
   ##
   # Extracts a IIIF info.json-like hash for an image
   #
-  # @example 
+  # @example
   #   ImageInfoExtractor.extract(my_image) # => {:image => info}
   #
   class ImageInfoExtractor
+
+    autoload :Jp2Extractor, 'iiif/image_info_extractor/jp2_extractor'
+
+    CONTEXT_URI = "http://iiif.io/api/image/2/context.json"
+    COMPLIANCE_URI = "http://iiif.io/api/image/2/level2.json"
+    PROTOCOL_URI = "http://iiif.io/api/image"
+
     ##
     # @param [Image] image  an image to gather info for
     # @return [Hash] the image info
     def self.extract(image)
-      {}
+      info_seed = {
```

Comment:
These constants should be ... constant ... but this current setup makes the actual extractor classes less clear. I'll keep working on those for now, and we can decide later if we want a separate class. I think we will because there's other stuff about the image that it would be nice to be able cache that does go in the info.json: the source_format and the embbeded color profile (if any) in particular. Associating those with the ImageInfo class, but still being able to send it `to_json` seems useful.